service/src/test/java/com/epam/esm/GiftCertificateTest.java

```diff
@@ -0,0 +1,186 @@
+package com.epam.esm;
+
+
+import com.epam.esm.converter.GiftCertificateConverter;
+import com.epam.esm.converter.TagConverter;
+import com.epam.esm.exception.DaoException;
+import com.epam.esm.exception.ServiceException;
+import com.epam.esm.impl.GiftCertificateServiceImpl;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.model.GiftCertificateDTO;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GiftCertificateTest {
+
+    @Mock
+    private GiftCertificateRepository giftCertificateRepository;
+    @Mock
+    private GiftCertificateConverter giftCertificateConverter;
+    @Mock
+    private TagRepository tagRepository;
+    @Mock
+    private TagConverter tagConverter;
+
+    private GiftCertificateService giftCertificateService;
+
+    @Before
+    public void init() {
+        giftCertificateService = new GiftCertificateServiceImpl(giftCertificateRepository, giftCertificateConverter, tagRepository, tagConverter);
+    }
+
+    @Test
+    public void shouldGetGiftCertificateDTOListForFindAllMethodCall() {
+        GiftCertificate giftCertificate = new GiftCertificate();
+        giftCertificate.setName("certificate");
+        giftCertificate.setDescription("description");
+        giftCertificate.setDuration(1);
+        giftCertificate.setPrice(new BigDecimal(10));
+        giftCertificate.setCreationDate(LocalDateTime.now());
+        giftCertificate.setLastUpdateDate(LocalDateTime.now());
+        GiftCertificateDTO giftCertificateDTO = new GiftCertificateDTO();
+        giftCertificateDTO.setName("certificate");
+        giftCertificateDTO.setDescription("description");
+        giftCertificateDTO.setDuration(1);
+        giftCertificateDTO.setPrice(new BigDecimal(10));
+        giftCertificateDTO.setCreationDate(LocalDateTime.now());
+        giftCertificateDTO.setLastUpdateDate(LocalDateTime.now());
```

Comment:
these lines should be moved to a separate method, cause it can be reused in other tests